composer.json

```diff
@@ -20,14 +20,14 @@
 	],
 	"require": {
 		"php": "~7.1",
-		"symfony/console": "~3.0",
-		"symfony/yaml": "~3.0"
+		"symfony/console": "~3.2|~4.0",
+		"symfony/yaml": "~3.0|~4.0"
 	},
 	"require-dev": {
-		"consistence/coding-standard": "2.0",
-		"jakub-onderka/php-parallel-lint": "0.9.2",
-		"phpstan/phpstan-shim": "0.8.4",
-		"phpunit/phpunit": "6.3.0"
+		"consistence/coding-standard": "^2.0",
```

Comment:
What about locking dev dependencies to specific versions to prevent "random" build failures?

Reply:
Well, at least you will fix it ASAP, no :)?

Its up to you, but I prefer not to worry about the upgrades, and just make sure it works (even regarding dev dependencies) with latest and lowest version by the constrains.

Reply:
The downside is that the build may break when you need to fix something small (and unrelated) in the project in a few months.

Reply:
What we do is to set [Cron job](https://docs.travis-ci.com/user/cron-jobs/) which rebuild the master each day, so if something breaks, we know it soon and make sure it is fixed ASAP.

But it depends on your development process. I have no problem to fixing the versions if you say so :).

Reply:
I think it is better to use the fixed versions here, as the project will probably be changed just rarely.

Reply:
But the minor version should not change and break anything :).

However changed ;)